### PR TITLE
[SPARK-43519][BUILD][SQL] Bump Parquet to 1.13.1

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

Upgrade Parquet from 1.13.0 to 1.13.1

### Why are the changes needed?

[Apache Parquet 1.13.1](https://parquet.apache.org/blog/2023/05/18/1.13.1/) is available, the release notes are

> ### Version 1.13.1 ###
> 
> Release Notes - Parquet - Version 1.13.1
> 
> #### Improvement
> 
> *   [PARQUET-2276](https://issues.apache.org/jira/browse/PARQUET-2276) - Bring back support > for Hadoop 2.7.3
> *   [PARQUET-2297](https://issues.apache.org/jira/browse/PARQUET-2297) - Skip delta problem > check
> *   [PARQUET-2292](https://issues.apache.org/jira/browse/PARQUET-2292) - Improve default > SpecificRecord model selection for Avro `{Write,Read}`Support
> *   [PARQUET-2290](https://issues.apache.org/jira/browse/PARQUET-2290) - Add CI for Hadoop 2
> *   [PARQUET-2282](https://issues.apache.org/jira/browse/PARQUET-2282) - Don't initialize HadoopCodec
> *   [PARQUET-2283](https://issues.apache.org/jira/browse/PARQUET-2283) - Remove Hadoop HiddenFileFilter
> *   [PARQUET-2081](https://issues.apache.org/jira/browse/PARQUET-2081) - Fix support for rewriting files without ColumnIndexes

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass GA

### CODE REVIEW

**Comment by pan3793** [comment]
cc @wangyum

File: docs/sql-data-sources-parquet.md [hunk 1]
```diff
@@ -350,7 +350,7 @@ Dataset<Row> df2 = spark.read().parquet("/path/to/table.parquet.encrypted");
...
 #### KMS Client
 
-The InMemoryKMS class is provided only for illustration and simple demonstration of Parquet encryption functionality. **It should not be used in a real deployment**. The master encryption keys must be kept and managed in a production-grade KMS system, deployed in user's organization. Rollout of Spark with Parquet encryption requires implementation of a client class for the KMS server. Parquet provides a plug-in [interface](https://github.com/apache/parquet-mr/blob/1.13.0/parquet-hadoop/src/main/java/org/apache/parquet/crypto/keytools/KmsClient.java) for development of such classes,
+The InMemoryKMS class is provided only for illustration and simple demonstration of Parquet encryption functionality. **It should not be used in a real deployment**. The master encryption keys must be kept and managed in a production-grade KMS system, deployed in user's organization. Rollout of Spark with Parquet encryption requires implementation of a client class for the KMS server. Parquet provides a plug-in [interface](https://github.com/apache/parquet-mr/blob/1.13.1/parquet-hadoop/src/main/java/org/apache/parquet/crypto/keytools/KmsClient.java) for development of such classes,
```
Comment (wangyum) [inline docs/sql-data-sources-parquet.md:353]: `1.13.1` -> `apache-parquet-1.13.1`?
https://github.com/apache/parquet-mr/blob/apache-parquet-1.13.1/parquet-hadoop/src/main/java/org/apache/parquet/crypto/keytools/KmsClient.java

Comment (pan3793) [reply docs/sql-data-sources-parquet.md:353]: Thanks for checking, fixed.

**Review by wangyum — APPROVED** [review]

**Comment by wangyum** [comment]
@pan3793 Could you update the pr description?

**Comment by dongjoon-hyun** [comment]
+1 for @wangyum 's comment.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by pan3793** [comment]
@wangyum @dongjoon-hyun, the PR description is updated, please take a look again.

**Comment by dongjoon-hyun** [comment]
Thank you for updating. Merged to master for Apache Spark 3.5.0.

**Pull request closed without merging**